src/platform/user/profile/vet360/components/AddressField/AddressField.jsx

```diff
@@ -135,6 +157,224 @@ export const convertCleanDataToPayload = (cleanData, fieldName) =>
     },
     e => !!e,
   );
+};
+
+const formSchema = {
+  type: 'object',
+  properties: {
+    'view:livesOnMilitaryBase': {
+      type: 'boolean',
+    },
+    'view:livesOnMilitaryBaseInfo': {
+      type: 'object',
+      properties: {},
+    },
+    countryName: {
+      type: 'string',
+      enum: ADDRESS_FORM_VALUES.COUNTRIES,
+    },
+    addressLine1: {
+      type: 'string',
+      minLength: 1,
+      maxLength: 100,
+      pattern: '^.*\\S.*',
+    },
+    addressLine2: {
+      type: 'string',
+      minLength: 1,
+      maxLength: 100,
+      pattern: '^.*\\S.*',
+    },
+    addressLine3: {
+      type: 'string',
+      minLength: 1,
+      maxLength: 100,
+      pattern: '^.*\\S.*',
+    },
+    city: {
+      type: 'string',
+    },
+    stateCode: {
+      type: 'string',
+      enum: Object.keys(ADDRESS_DATA.states),
+      enumNames: Object.values(ADDRESS_DATA.states),
+    },
+    province: {
+      type: 'string',
+    },
+    zipCode: {
+      type: 'string',
+      pattern: '^\\d{5}$',
+    },
+    internationalPostalCode: {
+      type: 'string',
+    },
+  },
+  required: ['countryName', 'addressLine1', 'city'],
+};
+
+const uiSchema = {
+  'view:livesOnMilitaryBase': {
+    'ui:title':
+      'I live on a United States military base outside of the United States',
+  },
+  'view:livesOnMilitaryBaseInfo': {
+    'ui:description': () => (
+      <div className="vads-u-padding-x--2p5">
+        <AdditionalInfo
+          status="info"
+          triggerText="Learn more about military base addresses"
+        >
+          <span>
+            The United States is automatically chosen as your country if you
+            live on a military base outside of the country.
+          </span>
+        </AdditionalInfo>
+      </div>
+    ),
+  },
+  countryName: {
+    'ui:title': 'Country',
+    'ui:options': {
+      updateSchema: formData => {
+        if (formData['view:livesOnMilitaryBase']) {
+          return {
+            enum: [USA.COUNTRY_NAME],
+          };
+        }
+        return {
+          enum: ADDRESS_FORM_VALUES.COUNTRIES,
+        };
+      },
+      updateUiSchema: formData => {
```

Comment:
Support for this new `udpateUiSchema` hook is added here 
![image](https://user-images.githubusercontent.com/20728956/73681899-1a52b500-4674-11ea-838d-435b6d0787b8.png)